src/lightning_habana/fabric/plugins/precision.py

```diff
@@ -12,66 +12,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Literal, Optional, Union, cast
+from contextlib import contextmanager
+from typing import Generator, Literal
 
+import torch
 from lightning_utilities import module_available
 from typing_extensions import get_args
 
 if module_available("lightning"):
     from lightning.fabric.plugins.precision.precision import Precision
-    from lightning.fabric.utilities.rank_zero import rank_zero_info
 elif module_available("pytorch_lightning"):
     from lightning_fabric.plugins.precision.precision import Precision
-    from lightning_fabric.utilities.rank_zero import rank_zero_info
 else:
     raise ModuleNotFoundError("You are missing `lightning` or `pytorch-lightning` package, please install it.")
 
 
-from lightning_habana import _HPU_AVAILABLE
-from lightning_habana.utils.imports import _HABANA_FRAMEWORK_AVAILABLE
-
-if _HABANA_FRAMEWORK_AVAILABLE:
-    from habana_frameworks.torch.hpex import hmp
-
-_PRECISION_INPUT_INT = Literal[32]
-_PRECISION_INPUT_STR = Literal["32", "bf16", "32-true", "bf16-mixed"]
-_PRECISION_INPUT = Union[_PRECISION_INPUT_INT, _PRECISION_INPUT_STR]
+_PRECISION_INPUT = Literal["32", "bf16", "32-true", "bf16-mixed"]
```

Comment:
Shall these be with `?-true` sufix?

Reply:
might be problematic when other types are supported & untested

Reply:
@Borda - can we close this ?